controllers/vmware/vspherecluster_reconciler.go

```diff
@@ -112,7 +112,8 @@ func (r ClusterReconciler) Reconcile(_ goctx.Context, req ctrl.Request) (_ ctrl.
 
 	// Handle deleted clusters
 	if !vsphereCluster.DeletionTimestamp.IsZero() {
-		return r.reconcileDelete(clusterContext), nil
+		r.reconcileDelete(clusterContext)
```

Comment:
Not sure what's going on. The linter was complaining about these finding on release-1.6 and release-1.6 but not on this branch.

I made the changes here as well so that main + all the supported release branches are in sync